### PR TITLE
Indy sdk add list

### PR DESCRIPTION
**Pull request by sschulz-t**
This pr  adds a missing definition for listMyDidsWithMeta (was listed as todo in the indy-sdk-tests.ts).

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/hyperledger/indy-sdk/blob/main/wrappers/nodejs/README.md
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## indy-sdk ([<kbd>unpkg</kbd>](https://unpkg.com/browse/indy-sdk@latest/))
was missing the following properties:
1. capi
1. setRuntimeConfig
1. issuerRotateCredentialDefStart
1. issuerRotateCredentialDefApply
1. proverSearchCredentials

<details>
<summary>as well as these 68 other properties...</summary>
<p>proverFetchCredentials, proverCloseCredentialsSearch, updateRevocationState, toUnqualified, setKeyMetadata, getKeyMetadata, cryptoAuthCrypt, cryptoAuthDecrypt, cryptoAnonDecrypt, replaceKeysStart, replaceKeysApply, storeTheirDid, keyForDid, setEndpointForDid, getEndpointForDid, setDidMetadata, getDidMetadata, getMyDidWithMeta, qualifyDid, submitAction, multiSignRequest, buildGetDdoRequest, buildNodeRequest, buildGetValidatorInfoRequest, buildGetTxnRequest, buildPoolConfigRequest, buildPoolRestartRequest, buildPoolUpgradeRequest, buildAuthRuleRequest, buildAuthRulesRequest, buildGetAuthRuleRequest, buildTxnAuthorAgreementRequest, buildDisableAllTxnAuthorAgreementsRequest, buildAcceptanceMechanismsRequest, appendRequestEndorser, getResponseMetadata, deleteWalletRecordTags, getSchema, getCredDef, purgeSchemaCache, purgeCredDefCache, isPairwiseExists, createPairwise, listPairwise, getPairwise, setPairwiseMetadata, createPaymentAddress, listPaymentAddresses, addRequestFees, parseResponseWithFees, buildGetPaymentSourcesRequest, buildGetPaymentSourcesWithFromRequest, parseGetPaymentSourcesResponse, parseGetPaymentSourcesWithFromResponse, buildPaymentReq, parsePaymentResponse, preparePaymentExtraWithAcceptanceData, buildMintReq, buildSetTxnFeesReq, buildGetTxnFeesReq, parseGetTxnFeesResponse, buildVerifyPaymentReq, parseVerifyPaymentResponse, getRequestInfo, signWithAddress, verifyWithAddress, refreshPoolLedger, listPools</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3043506d7d8eb0551f58046484b8a097bab73cd3
</p>

File: types/indy-sdk/index.d.ts [hunk 1]
```diff
@@ -708,4 +709,11 @@ export interface GetNymResponse {
     role: NymRole;
 }
 
+export interface DidWithMeta {
+    did: Did;
+    verkey: Verkey;
+    metadata: string;
+    tempVerkey: Verkey;
```
Comment (TimoGlastra) [inline types/indy-sdk/index.d.ts:716]: I don't see `tempVerkey` mentioned in the docs? 

https://github.com/hyperledger/indy-sdk/tree/main/wrappers/nodejs#listmydidswithmeta--wh----dids

Is it always returned?

Comment (sschulz-t) [reply types/indy-sdk/index.d.ts:716]: I took the info from the indy-sdk rust code (the nodejs code is a wrapper around this implementation):
https://github.com/hyperledger/indy-sdk/blob/1c7096dd95d0fd53881070f66907df4b9e61b874/libindy/src/commands/did.rs#L355

It is also in the nodejs test cases:  
https://github.com/hyperledger/indy-sdk/blob/1c7096dd95d0fd53881070f66907df4b9e61b874/wrappers/nodejs/test/did.js#L17

Comment (TimoGlastra) [reply types/indy-sdk/index.d.ts:716]: But it does seem that both  `metdata` and `tempVerkey` are optional, https://github.com/hyperledger/indy-sdk/blob/1c7096dd95d0fd53881070f66907df4b9e61b874/libindy/src/domain/crypto/did.rs#L161-L166

Which for the Node.JS wrapper means they can be null I think?

Comment (sschulz-t) [reply types/indy-sdk/index.d.ts]: Good find! When i compare
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bbff4b3696828af59cf6ae9132c47b4774df702d/types/indy-sdk/index.d.ts#L485
and

https://github.com/hyperledger/indy-sdk/blob/1c7096dd95d0fd53881070f66907df4b9e61b874/libindy/src/domain/anoncreds/credential.rs#L38

I updated my pr accordingly.

**Review by TimoGlastra — APPROVED** [review]

**Comment by TimoGlastra** [comment]
Ready to merge

**Comment by TimoGlastra** [comment]
Ready to merge

**Pull request merged**